report_dynamic/models/report_dynamic_section.py

```diff
@@ -0,0 +1,217 @@
+            self.copyvalue = "${{object.{} or {}}}".format(
+                self.field_id.name, self._get_proper_default_value()
+            )
+            self.sub_model_object_field_id = False
+        if self.field_id and self.field_id.relation:
+            self.sub_object_id = self.env["ir.model"].search(
+                [("model", "=", self.field_id.relation)]
+            )[0]
+        if self.sub_model_object_field_id:
+            self.copyvalue = "${{object.{}.{} or {}}}".format(
+                self.field_id.name,
+                self.sub_model_object_field_id.name,
+                self._get_proper_default_value(),
+            )
+
+    # this then needs to be an onchange, since user
+    # should be able to see preview after setting
+    # a condition, directly.
+    @api.onchange("condition_python")
+    def _compute_condition_python_preview(self):
+        """Compute condition and preview"""
+        for this in self:
+            this.condition_python_preview = False
+            if not (this.resource_ref_model_id and this.res_id and this.resource_ref):
+                continue
+            try:
+                # Check if there are any syntax errors etc
+                this.condition_python_preview = this._eval_condition_python()
+            except Exception as e:
+                # and show debug info
+                this.condition_python_preview = str(e)
+                continue
+
+    def _eval_condition_python(self):
+        if not self.condition_python:
+            return True
+        condition_python = (self.condition_python or "").strip()
+        record = self.resource_ref
+        if not record:
+            return False
+        return safe_eval(condition_python, {"object": record})
+
+    def _eval_condition_domain(self):
+        condition_domain = (self.condition_domain or "[]").strip()
+        return self.resource_ref.filtered_domain(safe_eval(condition_domain))
+
+    def _get_proper_default_value(self):
+        self.ensure_one()
+        is_num = self.field_id.ttype in ("integer", "float")
+        value = 0 if is_num else "''"
+        if self.default_value:
+            if is_num:
+                value = "{}"
+            else:
+                value = "'{}'"
+            value = value.format(self.default_value)
+        return value
+
+    # compute the dynamic content for jinja expression
+    def _compute_dynamic_content(self):
+        # a parent with two children
+        h = self._get_header_object()
+        for this in self:
+            if not (this._eval_condition_python() or this._eval_condition_domain()):
+                this.dynamic_content = ""
+                continue
+            prerendered_content = this._prerender()
+            content = this._render_template(
+                prerendered_content,
+                this.resource_ref_model_id.model,
+                this.res_id,
+                datas={"h": h},
+            )
```

Comment:
If any of this now throws an Exception, i still get an Internal Server Error on my preview.

If I however wrap this whole block inside:

```
            try:
                 # super-error-prone content generation
            except Exception:
                this.dynamic_content = "<pre>%s</pre>" % (
                    traceback.format_exc()
                )
                pass
```

I get shown the traceback direct in my preview and I know what to do to fix:

![image](https://user-images.githubusercontent.com/1466356/153233416-78e7da63-667d-45b7-ae7a-06c2b4e5e377.png)


https://stackoverflow.com/q/3702675